docs/content_credentials.md

```diff
@@ -0,0 +1,95 @@
+
+For Builder archives - Sign and save a manifest, either embedded or sidecar.  If you want to continue editing a saved manifest, then use Reader to read it, and then use reader.to_builder() (or Builder::from_reader?()) to convert it back into an editable manifest.
+
+- Note that this will work when reading any content. You can always extract an ingredient to a Builder.
+
+
+- The signature here can be local and does not need to be on a trust list. The purpose is to be able to detect any tampering of the data. But you could also use a trusted signature.
+
+
+- You can store the c2pa data as a .c2pa sidecar without modifying the source. The dest file can be a data sink.
+
+- signing the bits of the source image and capturing a thumbnail are optional.
+
+### new methods
+
+```
+    let builder = reader.to_builder()?;
+
+    builder.add_ingredient_from_reader(&reader)?;
+```
+
+### Questions
+
+- Should we allow any Reader to be converted to a Builder, or only those with the same claim_generator? Maybe there is some other flag.
+
+- Can we save a .c2pa file that does not need to validate against an asset? This should probably be an option, but lets think about it.
+
+### Test cases
+
+1) Validate an ingredient without a manifest, store in Builder and save.
+2) Validate an ingredient with a manifest, store in Builder and save.
+3) Read and display these saved Builders
+4) Create new Builder, add component Ingredient from Reader from asset.
+5) Add two ingredients to a builder, save, read, and then add both to new Builder and save/read.
+6) Create Builder and add opened Ingredient, save, read and convert to a Builder, add componentOf Ingredient, sign
+
+## Content Credentials API
+
+A future lower level API will wrap the Claim and Store structures providing a single object interface for C2PA. This would have the same underlying JUMBF object, but would not provide the declarative JSON api. It would also not have the higher level Ingredient and Resource APIs. Assertions would be added directly to a the content_credential and read directly from it. This will include direct access to CBOR and Binary assertions. Ingredient assertions would be built by first adding binary assertions for thumbnails and icons and then adding those references directly to the a Ingredient assertion before adding it. Content Credentials can be added to Content Credentials as Ingredients and Ingredient read as another Credential. In this model there are no Builder, Reader or Ingredient objects.
+
+- Content Credentials will always be created with Settings and be the common context.
```

Comment:
It could also be created with an HttpResolver or maybe a generic resource resolver as well? I think settings can be optional if you just want the defaults. That brings the question if we should have a builder for a `ContentCredential` (I think yes). It would allow you to "build" the settings as well.